src/Template/Element/sentences/sentence_and_translations.ctp

```diff
@@ -11,31 +31,31 @@ if (!isset($menuExpanded)) {
     $menuExpanded = false;
 }
 
-list($directTranslations, $indirectTranslations) = $translations;
-$maxDisplayed = 5;
-$showExtra = '';
-$numExtra = count($directTranslations) + count($indirectTranslations) - $maxDisplayed;
-$sentenceUrl = $this->Url->build(array(
-    'controller' => 'sentences',
-    'action' => 'show',
-    $sentence->id
-));
-$notReliable = $sentence->correctness == -1;
-
-$username = $user ? $user->username : null;
-$sentenceMenu = [
-    'canEdit' => CurrentUser::canEditSentenceOfUser($username),
-    'canReview' => CurrentUser::get('settings.users_collections_ratings'),
-    'canAdopt' => CurrentUser::canAdoptOrUnadoptSentenceOfUser($user),
-    'canDelete' => CurrentUser::canRemoveSentence($sentence->id, null, $username),
-    'canLink' => CurrentUser::isTrusted(),
-];
+if (isset($translations)) {
+    list($directTranslations, $indirectTranslations) = $translations;
+} else {
+    $directTranslations = $indirectTranslations = [];
+}
+
 $langs = $this->Languages->profileLanguagesArray(false, false);
 
-$userLanguagesJSON = htmlspecialchars(json_encode($langs), ENT_QUOTES, 'UTF-8');
-$sentenceJSON = $this->Sentences->sentenceForAngular($sentence);
-$directTranslationsJSON = $this->Sentences->translationsForAngular($directTranslations);
-$indirectTranslationsJSON = $this->Sentences->translationsForAngular($indirectTranslations);
+if (!isset($userLanguagesData)) {
+    $userLanguagesData = htmlspecialchars(json_encode($langs), ENT_QUOTES, 'UTF-8');
```

Comment:
By the way, you can use [`h($var)`](https://book.cakephp.org/3/en/core-libraries/global-constants-and-functions.html#h) instead of `htmlspecialchars($var, ENT_QUOTES, 'UTF-8')`.